src/Api/ResourceClassResolver.php

```diff
@@ -50,7 +50,7 @@ public function getResourceClass($value, string $resourceClass = null, bool $str
         }
 
         if (($strict && isset($type) && $resourceClass !== $type) || !$this->isResourceClass($typeToFind)) {
-            if (is_subclass_of($type, $resourceClass) && $this->isResourceClass($resourceClass)) {
+            if ($this->isResourceClass($resourceClass)) {
```

Comment:
It looks not OK to me. We lose information, we don't want only a resource here, but a resource of the given type.